### DISCUSSION TITLE
Use "with" syntax for nested elements

### DISCUSSION THREAD

**Discussion by dridk** [idom-team/idom#1230] [Ideas]
Did you seen the [niceGUI project ](https://nicegui.io/)? They use a very clever syntax to create the DOM syntax. 

Instead of using this syntax : 

```python
html.div(
    html.h1("My Todo List"),
    html.ul(
        html.li("Design a cool new app"),
        html.li("Build it"),
        html.li("Share it with the world!"),
    )
)
```

why not using this one   ! Easier to write and read

```python

with html.div():
    html.h1("My Todo List")
    with html.ul():
        html.li("Design a cool new app")
        html.li("Built it")
        html.lit("Share it with the word!")

```

**Comment by rmorshea** [comment]
This was debated a little bit here: https://github.com/reactive-python/reactpy/discussions/916#discussioncomment-5134918

Technically you could create your own utility that implements this interface.

**Comment by Archmonger** [comment]
With the release of ReactPy v2 rapidly approaching, I've created a new ticket about this #1279.

Hopefully we can get some users that voice their opinion on which interface feels better to use.
